scene/gui/layout_container.cpp

```diff
@@ -0,0 +1,236 @@
+
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "preset",
+						 PROPERTY_HINT_ENUM, "Top Left,Top Right,Bottom left,Bottom Right,Center left,Center Top,Center Right,Center Bottom,Center,Left Wide,Top Wide,Right Wide,Bottom Wide,Vertical Center Wide,Horizontal Center Wide,Full Rect,Custom", PROPERTY_USAGE_DEFAULT),
+			"set_margin_layout_preset", "get_margin_layout_preset");
+
+	ADD_GROUP("Custom relative margins", "custom_relative_margin_");
+	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "custom_relative_margin_left", PROPERTY_HINT_RANGE, "0,1,0.001"), "_set_custom_relative_margin", "get_custom_relative_margin", SIDE_LEFT);
+	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "custom_relative_margin_top", PROPERTY_HINT_RANGE, "0,1,0.001"), "_set_custom_relative_margin", "get_custom_relative_margin", SIDE_TOP);
+	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "custom_relative_margin_right", PROPERTY_HINT_RANGE, "0,1,0.001"), "_set_custom_relative_margin", "get_custom_relative_margin", SIDE_RIGHT);
+	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "custom_relative_margin_bottom", PROPERTY_HINT_RANGE, "0,1,0.001"), "_set_custom_relative_margin", "get_custom_relative_margin", SIDE_BOTTOM);
+
+	ADD_GROUP("Margins", "margin_");
+	ADD_PROPERTYI(PropertyInfo(Variant::INT, "margin_left", PROPERTY_HINT_RANGE, "-4096,4096,suffix:px"), "set_margin", "get_margin", SIDE_LEFT);
+	ADD_PROPERTYI(PropertyInfo(Variant::INT, "margin_top", PROPERTY_HINT_RANGE, "-4096,4096,suffix:px"), "set_margin", "get_margin", SIDE_TOP);
+	ADD_PROPERTYI(PropertyInfo(Variant::INT, "margin_right", PROPERTY_HINT_RANGE, "-4096,4096,suffix:px"), "set_margin", "get_margin", SIDE_RIGHT);
+	ADD_PROPERTYI(PropertyInfo(Variant::INT, "margin_bottom", PROPERTY_HINT_RANGE, "-4096,4096,suffix:px"), "set_margin", "get_margin", SIDE_BOTTOM);
+
+	BIND_ENUM_CONSTANT(PRESET_TOP_LEFT);
+	BIND_ENUM_CONSTANT(PRESET_TOP_RIGHT);
+	BIND_ENUM_CONSTANT(PRESET_BOTTOM_LEFT);
+	BIND_ENUM_CONSTANT(PRESET_BOTTOM_RIGHT);
+	BIND_ENUM_CONSTANT(PRESET_CENTER_LEFT);
+	BIND_ENUM_CONSTANT(PRESET_CENTER_TOP);
+	BIND_ENUM_CONSTANT(PRESET_CENTER_RIGHT);
+	BIND_ENUM_CONSTANT(PRESET_CENTER_BOTTOM);
+	BIND_ENUM_CONSTANT(PRESET_CENTER);
+	BIND_ENUM_CONSTANT(PRESET_LEFT_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_TOP_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_RIGHT_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_BOTTOM_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_VCENTER_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_HCENTER_WIDE);
+	BIND_ENUM_CONSTANT(PRESET_FULL_RECT);
+	BIND_ENUM_CONSTANT(PRESET_CUSTOM);
+
+	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, LayoutContainer, margin_left);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, LayoutContainer, margin_top);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, LayoutContainer, margin_right);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, LayoutContainer, margin_bottom);
+}
+
+Size2 LayoutContainer::get_minimum_size() const {
+	Size2 ms;
+	for (int i = 0; i < get_child_count(); i++) {
+		Control *c = as_sortable_control(get_child(i));
```

Comment:
```suggestion
		Control *c = as_sortable_control(get_child(i), SortableVisbilityMode::VISIBLE);
```
This uses `is_visible()` instead of `is_visible_in_tree()` when determining child Control. I don't know if it's necessary here, but it helped in some containers.